lib/nanoc/base/views/item.rb

```diff
@@ -31,10 +31,11 @@ def identifier
 
     # @see Hash#fetch
     def fetch(key, fallback = NONE, &_block)
-      res = @item[key] # necessary for dependency tracking
+      Nanoc::Int::NotificationCenter.post(:visit_started, @item)
+      Nanoc::Int::NotificationCenter.post(:visit_ended,   @item)
```

Comment:
The “visit” ends before the method returns. There’s nothing to be done _during_ the visit in this case.

There’s really only one place where stuff needs to be tracked during a visit (between start/end) and that is when compiling an item.

Having a `:visited` shorthand could be useful, I guess!